### PR TITLE
Replace grams and milliliters with a simple amoun-in-grams representation

### PR DESCRIPTION
**Pull request by andreas-bauer**

### CODE REVIEW

**Comment by andreas-bauer** [comment]
Multi-language support will be added in an additional PR.

File: ui/src/views/RecipeUpsert.vue [hunk 1]
```diff
@@ -331,8 +215,8 @@ export default class RecipeUpsert extends Vue {
   scaleIngredients = true;
   valid = false;
 
-  availableVictuals: IngredientMagnifier[] = [];
-  selectedVictuals: IngredientMagnifier[] = [];
+  availableIngrediants: Ingredient[] = [];
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:218]: Typo: ingredi**e**nt

**Comment by georg-schwarz** [comment]
Phew.. a few points I noticed:
* is it really less complex than the solution before? (html yes, code no)
* victuals can only be handled in one amount type, right? 
* I didn't see any place where we calculate to grams.. So how will we calculate the nutriment information for a recipe? There we need the mapping from gram to the amount type because we always specify nutriments per 100g

Basically this is how I implemented in the older version. However, I noticed that it has its benefits if you can switch between e.g. pieces and grams depending on the recipe. But I think the unit most used is definitely grams for teachers. Everything is a bonus that won't be really used probably.

File: ui/src/api/RecipeApi.ts [hunk 2]
```diff
@@ -80,8 +79,7 @@ const pear: Victual = {
   allergens: [Allergens.celery, Allergens.agar_agar, Allergens.L_ascorbic_acid],
   categories: [VictualCategory.fruits],
   createdAt: new Date(Date.now()),
-  gramsPer100Ml: null,
-  gramsPerPiece: 60
+  amountInGrams: 204
```
Comment (georg-schwarz) [inline ui/src/api/RecipeApi.ts:82]: For what do we need that? I think we only need amount in ingredients (?)

File: ui/src/api/VictualApi.ts [hunk 3]
```diff
@@ -64,8 +64,7 @@ mockedVictualApi.create({
   allergens: [],
   categories: [VictualCategory.fruits],
   createdAt: new Date(Date.now()),
-  gramsPer100Ml: null,
-  gramsPerPiece: 60
+  amountInGrams: 150
```
Comment (georg-schwarz) [inline ui/src/api/VictualApi.ts:67]: Same here

File: ui/src/model/victual.ts [hunk 4]
```diff
@@ -5,8 +5,7 @@ export default interface Victual {
   allergens: Allergens[];
   categories: VictualCategory[];
   createdAt: Date;
-  gramsPerPiece: number | null;
-  gramsPer100Ml: number | null;
+  amountInGrams: number;
```
Comment (georg-schwarz) [inline ui/src/model/victual.ts:8]: Remove

File: ui/src/views/RecipeUpsert.vue [hunk 5]
```diff
@@ -76,25 +76,21 @@
 
                 <draggable
                   class="list-group"
-                  :list="filteredVictuals"
+                  :list="filteredIngrediants"
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:79]: ingredi**e**nts, not ingrediants

File: ui/src/views/RecipeUpsert.vue [hunk 6]
```diff
@@ -128,101 +124,36 @@
...
+                  <div
+                    v-for="item in selectedIngrediants"
+                    :key="item.victual.id"
+                  >
                     <v-divider></v-divider>
                     <v-list-item class="grab-curser">
                       <v-list-item-content>
                         <v-list-item-title
-                          v-text="item.ingredient.victual.name"
+                          v-text="item.victual.name"
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:138]: Are u sure that its not item.ingredient.victual.name?

File: ui/src/views/RecipeUpsert.vue [hunk 7]
```diff
@@ -377,37 +260,38 @@ export default class RecipeUpsert extends Vue {
...
-    this.selectedVictuals = [];
+    this.selectedIngrediants.forEach(e => this.filteredIngrediants.push(e));
+    this.selectedIngrediants = [];
   }
 
-  get filteredVictuals(): IngredientMagnifier[] {
-    return this.availableVictuals.filter(item => {
-      if (!this.search) return this.availableVictuals;
-      return item.ingredient.victual.name
+  get filteredIngrediants() {
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:291]: Add type information

File: ui/src/views/RecipeUpsert.vue [hunk 8]
```diff
@@ -420,8 +304,10 @@ export default class RecipeUpsert extends Vue {
   @Watch("recipe.servingSize")
   onServiceSizeChanged(val: number, oldVal: number) {
     if (this.scaleIngredients) {
-      for (const ingredientMagnifier of this.selectedVictuals) {
-        ingredientMagnifier.grams *= val / oldVal;
+      for (const ingredientMagnifier of this.selectedIngrediants) {
```
Comment (georg-schwarz) [inline ui/src/views/RecipeUpsert.vue:307]: rename to ingredient

**Comment by georg-schwarz** [comment]
* renamed Victual.amountInGrams to Victual.defaultAmountInGrams
* some minor refactorings (renamings)

Ready to merge from my side!

**Pull request merged**